source/base/polynomial.cc

```diff
@@ -1228,6 +1234,230 @@ namespace Polynomials
     return basis;
   }
 
+
+// ------------------ HermiteLikeInterpolation --------------- //
+  namespace
+  {
+    // Finds the zero position N such that the mass matrix entry (0,1) with
+    // the Hermite polynomials evaluates to zero. The function has originally
+    // been derived by a secant method for the integral entry l_0(x) * l_1(x)
+    // but we only need to do one iteration because the zero N is linear in
+    // the integral value.
+    double find_variable_support_point (const Vector<double> &eigval)
+    {
+      // initial guess for the support point position values
+      double N_left = 0;
+      double N_right = 0.5;
+      const unsigned int degree = eigval.size() + 3;
+
+      // compute the integrals of the product of l_0(x) * l_1(x)
+      // l_0(x) = (x-N)*(x-eigval(0))*...*(x-eigval(degree-3))*(x-1)*(x-1)
```

Comment:
We have not normalized yet at this point, and we don't have to anyway because all we do is to find the position of the root such that we get a zero for that particular matrix entry. So the polynomial really isn't normalized. I will add a short comment about this fact in the code.